### ISSUE TITLE
Treat constructor arguments

### ISSUE THREAD

**Issue by MrChico** [dapphub/dapptools#476]
Currently, we are not able to
a) symbolically execute init code with symbolic arguments
b) get srcmapping for init code with arguments

We could change `ContractCode` as follows:
```
data ContractCode
  = InitCode (ByteString, Buffer)    -- ^ "Constructor" code, during contract creation (possibly with arguments)
  | RuntimeCode ByteString           -- ^ "Instance" code, after contract creation
  deriving (Show, Eq)
``

**Comment by MrChico** [comment]
I think this problem was actually introduced in https://github.com/dapphub/dapptools/pull/334 which introduced a much "smarter" way of detecting the metadata, which fails to take constructor arguments into account

**Comment by MrChico** [comment]
b) was solved in #530 , a) still remains a problem

**Comment by MrChico** [comment]
solved in #626
